src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -645,33 +670,38 @@ void checkMemberStateOnConstructorExit(MethodSymbol constructor, Symbol member,
                     return;
                 }
 
-                TypeWithAnnotations fieldType;
+                TypeWithAnnotations symbolType;
                 FieldSymbol? field;
                 Symbol symbol;
                 switch (member)
                 {
                     case FieldSymbol f:
-                        fieldType = f.TypeWithAnnotations;
+                        symbolType = f.TypeWithAnnotations;
                         field = f;
                         symbol = (Symbol?)(f.AssociatedSymbol as PropertySymbol) ?? f;
                         break;
                     case EventSymbol e:
-                        fieldType = e.TypeWithAnnotations;
+                        symbolType = e.TypeWithAnnotations;
                         field = e.AssociatedField;
                         symbol = e;
                         if (field is null)
                         {
                             return;
                         }
                         break;
+                    case PropertySymbol p when forcePropertyAnalysis:
+                        symbolType = p.TypeWithAnnotations;
+                        field = null;
+                        symbol = p;
+                        break;
                     default:
                         return;
                 }
-                if (field.IsConst)
+                if (field?.IsConst ?? false)
```

Comment:
>if (field?.IsConst ?? false) [](http://example.com/codeflow?start=16&length=28)

Could this check move inside the `case FieldSymbol f:` case above? Then the `field` local can be removed.

Reply:
I'd rather not, as I think that makes the code a bit harder to follow. Right now, the switch explicitly sets up the things that need to be checked, and then we check them. This would blend those two steps.